subsys/power/policy/policy_residency.c

```diff
@@ -73,15 +73,20 @@ int sys_pm_policy_next_state(s32_t ticks, enum power_states *pm_state)
 	}
 
 	for (i = ARRAY_SIZE(pm_policy) - 1; i >= 0; i--) {
+#ifdef CONFIG_PM_CONTROL_STATE_LOCK
```

Comment:
```
if (IS_ENABLED(CONFIG_PM_CONTROL_STATE_LOCK))
```
maybe

Reply:
also in other places

Reply:
The  sys_pm_ctrl_is_state_enabled() is not declared when CONFIG_PM_CONTROL_STATE_LOCK != 1, so change to IS_ENABLED() will cause compilation error.